sktime/datasets/forecasting/m5_competition.py

```diff
@@ -93,9 +94,14 @@ def path_to_data_dir(self):
     def _download_if_needed(self):
         """Download the data if it is not already downloaded."""
         if not self.path_to_data_dir.exists():
```

Comment:
is this and other `if not path.exists` not already covered by the boilerplate in `download`? Can it be removed, or does it serve another purpose?